### PR TITLE
DEVDOCS-4671: [new] GraphQL Storefront API, add complexity response header

### PR DESCRIPTION
**Pull request by bc-andreadao**
# [DEVDOCS-4671]

Added complexity header for graphql responses

[DEVDOCS-4671]: https://bigcommercecloud.atlassian.net/browse/DEVDOCS-4671?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/api-docs/storefront/graphql/graphql-api-overview.md [hunk 1]
```diff
@@ -472,7 +472,9 @@ query brands {
 
 ## Complexity limits
 
-The GraphQL Storefront API uses an algorithm to calculate a complexity score for queries made against the API. Queries that exceed the complexity score will receive an error response:
+The GraphQL Storefront API uses an algorithm to calculate a complexity score for queries made against the API. The API returns the query complexity as an integer in the HTTP response header `x-bc-graphql-complexity` when you send requests that are at least partially successful (e.g. not rate limiting).
```
Comment (zvuki) [inline docs/api-docs/storefront/graphql/graphql-api-overview.md:475]:
```suggestion
The GraphQL Storefront API uses an algorithm to calculate a complexity score for queries made against the API. The API returns the query complexity as an integer in the HTTP response header `x-bc-graphql-complexity` when you send a valid GraphQL request.
```

This is probably more accurate — @bc-dima-pasieka @6juara9?

**Comment by bc-andreadao** [comment]
@zvuki Thanks for the suggestion. I talked to @6juara9 and it turns out only valid GraphQL requests have the header, so I committed your suggestion.

**Review by 6juara9 — DISMISSED** [review]

File: docs/api-docs/storefront/graphql/graphql-api-overview.md [hunk 2]
```diff
@@ -472,7 +472,9 @@ query brands {
 
 ## Complexity limits
 
-The GraphQL Storefront API uses an algorithm to calculate a complexity score for queries made against the API. Queries that exceed the complexity score will receive an error response:
+The GraphQL Storefront API uses an algorithm to calculate a complexity score for queries made against the API. The API returns the query complexity as an integer in the HTTP response header `x-bc-graphql-complexity` when you send a valid GraphQL request.
+
+Queries whose complexity score exceed the limit will receive an error response:
```
Comment (bc-traciporter) [inline docs/api-docs/storefront/graphql/graphql-api-overview.md:477]:
```suggestion
Queries whose complexity score exceeds the limit will receive the following error response:

```

**Review by bc-traciporter — COMMENTED** [review]
I made one suggestion.

File: docs/api-docs/storefront/graphql/graphql-api-overview.md (hunk 2 above)
Comment (stanzikratelbc) [inline docs/api-docs/storefront/graphql/graphql-api-overview.md:477]: @bc-andreadao I would change this sentence to say the following: "If the complexity score of a query exceeds the complexity limit, the following error code will be sent:"

Also, can the complexity limit be updated (like from 1000 to 2000 for example?)

**Review by stanzikratelbc — COMMENTED** [review]
@bc-andreadao Please see my comment about the complexity limit error message.

File: docs/api-docs/storefront/graphql/graphql-api-overview.md (hunk 2 above)
Comment (bc-andreadao) [reply docs/api-docs/storefront/graphql/graphql-api-overview.md]: @stanzikratelbc I copyedited to change the sentence structure and fix the grammar, but made it active voice

Thanks for bringing up the update for the limit. The attached screenshot that I just received from @6juara9 has a complexity limit of 10000, so I updated the doc.
complexity limit

**Review by bc-dima-pasieka — DISMISSED** [review]

File: docs/api-docs/storefront/graphql/graphql-api-overview.md [hunk 3]
```diff
@@ -472,12 +472,15 @@ query brands {
...
+The GraphQL Storefront API uses an algorithm to calculate a complexity score for queries made against the API. The API returns the query complexity as an integer in the HTTP response header `x-bc-graphql-complexity` when you send a valid GraphQL request.
+
+If a query's complexity score exceeds the complexity limit, you will receive the following error response:
+
 
 ```json title="Example response with complexity error"
 {
   "error": {
-    "error": "The query is too complex as it has a complexity score of 1223 out of 1000. Please remove some elements and try again"
+    "error": "The query is too complex as it has a complexity score of 12230 out of 10000. Please remove some elements and try again"
```
Comment (slsriehl) [inline docs/api-docs/storefront/graphql/graphql-api-overview.md:483]: is there a period at the end of this sentence when you receive this message back? wondering if this is verbatim.

File: docs/api-docs/storefront/graphql/graphql-api-overview.md [hunk 4]
```diff
@@ -472,12 +472,15 @@ query brands {
 
 ## Complexity limits
 
-The GraphQL Storefront API uses an algorithm to calculate a complexity score for queries made against the API. Queries that exceed the complexity score will receive an error response:
+The GraphQL Storefront API uses an algorithm to calculate a complexity score for queries made against the API. The API returns the query complexity as an integer in the HTTP response header `x-bc-graphql-complexity` when you send a valid GraphQL request.
+
+If a query's complexity score exceeds the complexity limit, you will receive the following error response:
```
Comment (slsriehl) [inline docs/api-docs/storefront/graphql/graphql-api-overview.md:477]:
```suggestion
If a query's complexity score exceeds the complexity limit, you will receive an error response similar to the following:

```

the score will be different each time. also the text might change a little in the future.

File: docs/api-docs/storefront/graphql/graphql-api-overview.md [hunk 5]
```diff
@@ -472,12 +472,15 @@ query brands {
 
 ## Complexity limits
 
-The GraphQL Storefront API uses an algorithm to calculate a complexity score for queries made against the API. Queries that exceed the complexity score will receive an error response:
+The GraphQL Storefront API uses an algorithm to calculate a complexity score for queries made against the API. The API returns the query complexity as an integer in the HTTP response header `x-bc-graphql-complexity` when you send a valid GraphQL request.
```
Comment (slsriehl) [inline docs/api-docs/storefront/graphql/graphql-api-overview.md:475]:
```suggestion
The GraphQL Storefront API uses an algorithm to calculate a complexity score for queries made against the API. When you send a valid GraphQL request, the API returns the query complexity as an integer in the HTTP response header `x-bc-graphql-complexity` .

```
condition before result. lead with the condition that the request is valid.

Comment (slsriehl) [reply docs/api-docs/storefront/graphql/graphql-api-overview.md:475]: also prepositions; can you find a better one than `in`? It's not quite right.

File: docs/api-docs/storefront/graphql/graphql-api-overview.md (hunk 3 above)
Comment (bc-andreadao) [reply docs/api-docs/storefront/graphql/graphql-api-overview.md:483]: No period, here's the screenshot https://github.com/bigcommerce/dev-docs/pull/1719#discussion_r1067456759

File: docs/api-docs/storefront/graphql/graphql-api-overview.md [hunk 6]
```diff
@@ -472,12 +472,17 @@ query brands {
 
 ## Complexity limits
 
-The GraphQL Storefront API uses an algorithm to calculate a complexity score for queries made against the API. Queries that exceed the complexity score will receive an error response:
+The GraphQL Storefront API uses an algorithm to calculate a complexity score for queries made against the API. When you send a valid GraphQL request, the API returns the query complexity as an integer using the HTTP response header `x-bc-graphql-complexity` .
+
```
Comment (slsriehl) [inline docs/api-docs/storefront/graphql/graphql-api-overview.md:476]:
```suggestion

```

File: docs/api-docs/storefront/graphql/graphql-api-overview.md [hunk 7]
```diff
@@ -472,12 +472,17 @@ query brands {
 
 ## Complexity limits
 
-The GraphQL Storefront API uses an algorithm to calculate a complexity score for queries made against the API. Queries that exceed the complexity score will receive an error response:
+The GraphQL Storefront API uses an algorithm to calculate a complexity score for queries made against the API. When you send a valid GraphQL request, the API returns the query complexity as an integer using the HTTP response header `x-bc-graphql-complexity` .
+
+
+If a query's complexity score exceeds the complexity limit, you will receive an error response similar to the following:
+
```
Comment (slsriehl) [inline docs/api-docs/storefront/graphql/graphql-api-overview.md:479]:
```suggestion

```

**Review by slsriehl — APPROVED** [review]
nice work.

**Comment by bc-andreadao** [comment]
@slsriehl I got the go ahead to merge (see [CSS-944](https://bigcommercecloud.atlassian.net/browse/CSS-944))

**Pull request merged**